### DISCUSSION TITLE
Official AlpineJS Devtools

### DISCUSSION THREAD

**Discussion by james0r** [alpinejs/alpine#3505] [3. Feature ideas]
[This](https://github.com/alpine-collective/alpinejs-devtools) is pretty great and does the job most all the time, but it looks like it hasn't had a release since before Alpine v3 and doesn't support Alpine stores, and possibly looks to be abandoned.

Would be awesome to see what it could do with the alpine team behind it.

**Comment by ekwoka** [comment]
It does still work though. I use it all the time.

Stores would be nice, but they are quite a bit harder to get into.

**Reply by james0r** [reply]
Oh yeah works great for components. Nothing but respect. Just think it would be a huge improvement to get stores in there.

**Comment by browner12** [comment]
The current devtools is good, but doesn't hold a candle to the Vue devtools.  And as already mentioned there doesn't seem to be active development on it. Would love to see it brought into the official organization.

I may have a sizable chunk of my budget clearing up at work in the near future, and I've been contemplating seeing if I can use some of it to grease the wheels on getting this off the ground. Will let people know if anything comes of it.

**Reply by ekwoka** [reply]
Would be fantastic!

Another thing that could be nice is having some way for headless components to hook in and tell the devtools that they exist and how to access the data they may contain.

Even if you use a directive to bind `x-data` the devtools do not see the data. Even more so if you keep the data outside of it.

**Reply by HugoDF** [reply]
👋 @ekwoka @james0r I know this is a while in the future

Is there any further info on Alpine.js providing a way to hook into store instances?

I'm seeing what I can do with devtools moving forward, feel free to fill out the survey at  https://github.com/alpine-collective/alpinejs-devtools/issues/431

**Reply by ekwoka** [reply]
```js
Alpine.$data(somealpineelement).$store
```

This SHOULD return the underlying store object.

Or just doing whatever to evaluate `$store` on an element

**Reply by HugoDF** [reply]
Wanted to say thanks for the above again, looks like it works for >=3.8.0

I've got stores working in devtools locally 🙂

**Reply by ekwoka** [reply]
To clarify why this works for anyone interested, while magics are injected into the scope of expressions, they are also injected onto the actual data context whenever one is made (with x-data), so you can just get the datastack at any element and use the magics on that object directly.

**Comment by james0r** [comment]
Would be super nice to be able to hide evaluated directives in the markup too so you can just see the resultant attributes and values, and possibly the reverse too.

**Reply by ekwoka** [reply]
That would probably be really hard to handle, simply because the mutation observers that handle cleaning up directives watches the actual dom for attribute changes.

I guess potentially with just `mutateDom` it could be done, but that may have side effects....
